src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
+                if (count == 0)
+                {
+                    await Task.Delay(100).ConfigureAwait(false);
```

Comment:
Changed the code so that polling is done on-demand. The very first request may still have a delay. The 100ms value is just a guess - not sure what else to use